### PR TITLE
PLATUI-1956: Initial commit

### PR DESCRIPTION
**Pull request by tmc-ee**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+object Dependencies {
+
+  val test: Seq[ModuleID] = Seq(
+    "com.typesafe"         % "config"         % "1.4.2"    % Test,
+    "com.vladsch.flexmark" % "flexmark-all"   % "0.62.2"   % Test,
+    "org.scalatest"       %% "scalatest"      % "3.2.13"   % Test,
+    "org.scalatestplus"   %% "selenium-4-2"   % "3.2.13.0" % Test,
+    "org.slf4j"            % "slf4j-simple"   % "2.0.0"    % Test,
+    "uk.gov.hmrc"         %% "webdriver-hmrc" % "1.0.0"    % Test
```
Comment (tmc-ee) [inline project/Dependencies.scala:11]: The defined version of `webdriver-hmrc` should be updated to use the `+` notation once we establish versioning for the [webdriver-hmrc](https://github.com/hmrc/webdriver-hmrc) library to enable automatic updates.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,58 @@
...
+
+```bash
+./run-docker-browser.sh <browser>
+```
+
+## Tests
+
+Run tests as follows:
+
+* Argument `<browser>` must be `remote-chrome`, `remote-edge`, or `remote-firefox`.
```
Comment (tmc-ee) [inline README.md:35]: The arguments specified for available browsers documented in the README should be updated once we change the name of the available browsers in [webdriver-hmrc](https://github.com/hmrc/webdriver-hmrc) from `remote-browser` to `browser`.

File: src/test/scala/uk/gov/hmrc/ui/utilities/TestConfiguration.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ui.utilities
+
+import com.typesafe.config.{Config, ConfigFactory}
+
+object TestConfiguration {
+
+  val config: Config        = ConfigFactory.load()
```
Comment (tmc-ee) [inline src/test/scala/uk/gov/hmrc/ui/utilities/TestConfiguration.scala:23]: The scope of these values should be private.

File: run-tests.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -e
+
+BROWSER=$1
+ENVIRONMENT=$2
+
+sbt scalafmtCheckAll scalafmtSbtCheck
+
+sbt -Dbrowser="${BROWSER:=remote-chrome}" -Denvironment="${ENVIRONMENT:=local}" "testOnly uk.gov.hmrc.ui.specs.*"
```
Comment (tmc-ee) [inline run-tests.sh:8]: The default browser argument specified should be updated once we change the name of the available browsers in [webdriver-hmrc](https://github.com/hmrc/webdriver-hmrc) from `remote-chrome` to `chrome`.

File: build.sbt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+lazy val root = (project in file("."))
+  .settings(
+    name := "platform-testing-example-ui-journey-tests",
+    version := "0.1.0-SNAPSHOT",
```
Comment (tmc-ee) [inline build.sbt:4]: Should the project version be a SNAPSHOT?

**Comment by tmc-ee** [comment]
Should page object class names be appended with `Page`?

File: src/test/scala/uk/gov/hmrc/ui/specs/BaseSpec.scala [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import org.scalatest.{BeforeAndAfterEach, GivenWhenThen, Outcome}
+import org.scalatestplus.selenium.WebBrowser
+import uk.gov.hmrc.webdriver.DriverFactory
+
+trait BaseSpec extends AnyFeatureSpec with BeforeAndAfterEach with GivenWhenThen with Matchers with WebBrowser {
+
+  implicit var driver: RemoteWebDriver = _
+
+  override def beforeEach(): Unit = {
+    driver = new DriverFactory().initialise()
```
Comment (tmc-ee) [inline src/test/scala/uk/gov/hmrc/ui/specs/BaseSpec.scala:32]: Should we provide an example of how to add custom browser options here, in the README or in the README of [webdriver-hmrc](https://github.com/hmrc/webdriver-hmrc)?

**Review by vivrichards600 — APPROVED** [review]

File: src/test/scala/uk/gov/hmrc/ui/specs/BaseSpec.scala (hunk 6 above)
Comment (tmc-ee) [reply src/test/scala/uk/gov/hmrc/ui/specs/BaseSpec.scala:32]: IMO this should be done as part of the guidance provided in the README of [webdriver-hmrc](https://github.com/hmrc/webdriver-hmrc).

**Review by vivrichards600 — APPROVED** [review]

**Pull request merged**